Ref #3 Add call button

ir4y opened the pull request.

ruscoder commented on src/containers/Chat/index.tsx:
```diff
@@ -146,11 +146,27 @@ export class Component extends React.Component<ComponentProps, {}> {
+                        },
+                    });
+                }
+                await CallService.startOutgoingCall(users[0].username, users[0].displayName);
```
@jasiemion `users` it is a list of all contacts. Now you call always to the first user. It is wrong. Try to use a username prop and find the user's display name in the users' list.

jussiemion replied:
@ruscoder ok.

jasiemion replied:
I didn't found username prop, only userId in conversation object. But problem has been fixed.

jasiemion replied:
Additional amendments: variables has been shortened, excess code has been replaced with method makeUsername.

jussiemion commented:
I didn't found username prop, only userId in conversation object. But problem has been fixed.

UPD. Additional amendments: variables has been shortened, excess code has been replaced with method makeUsername.

ruscoder commented on src/containers/Chat/index.tsx:
```diff
@@ -146,11 +146,34 @@ export class Component extends React.Component<ComponentProps, {}> {
-        if (buttonId === 'delete') {
+        const { conversation, session } = this.props;
+
+        const conversationalists = conversation.participants.filter(
```
You already have `session.username`. Why do you find a user with userId includes a username and make the username again from a userId?

You are doing a lot of unneeded operations. Try to fix your code.

Commit:
Create makeOutgoingCall method in Call service

Commit:
Ref #3 Add call button

Commit:
Extend .gitignore

Commit:
Rewrite calls with makeOutgoingCall from Call service

Commit:
Refactor Call service

- remove excess try ... catch construction
- split object with user data to username and displayName arguments

Commit:
Refactor logic of calls

- remove excess object with user data
- replace arguments in makeOutgoingCall methods
- rewrite try ... catch constructions with if ... else
- add missed isPending flag restoring

ruscoder commented on src/services/call.ts:
```diff
@@ -132,6 +134,25 @@ export class CallService {
+        try {
+            await this.requestPermissions(false);
+            const { call } = await this.startOutgoingCall(username, displayName);
+            const result = success(call);
```
You forgot about cursor.set(result) here

ruscoder commented on src/containers/ContactList/index.tsx:
```diff
@@ -109,20 +111,22 @@ export class Component extends React.Component<ComponentProps, {}> {
+            user.displayName
+        );
+
+        if (isSuccess(callResponse)) {
```
Simplify it using:
```
if (isFailure(callResponse)) {
... show modal
}

this.props.tree.isPending.set(false);
```

ruscoder commented on src/containers/Chat/index.tsx:
```diff
@@ -151,13 +153,39 @@ export class Component extends React.Component<ComponentProps, {}> {
+
             this.props.tree.isPending.set(true);
+
             try {
```
Get rid of `try` block. Rewrite it in the same way as you did it in Call screen

Commit:
Remove ramda and ramda-adjunct dependencies

Commit:
Remove ramda and ramda-adjunct dependencies

Commit:
Refactor Call service and calls logic

- call.ts: put missed pushing result into cursor
- contactList/index.ts: simplify conditional construct
- Chat/index.ts: replace try ... catch costruction with conditional
construct

Commit:
Merge branch 'develop' into issue-3

ruscoder commented on src/containers/Chat/index.tsx:
```diff
@@ -387,7 +412,7 @@ export class Component extends React.Component<ComponentProps, {}> {
         return (
             <FlatList<ChatMessage>
-                data={R.reverse(messages)}
+                data={_.reverse(messages.slice())}
```
`_.reverse(_.clone(messages))`

ruscoder commented on src/services/chat.ts:
```diff
@@ -120,29 +120,24 @@ export async function createConversation(
+            const conversations = await Promise.all(_.map(myUser.conversationsList, catchConversation));
+            const existingConversation = _.find(conversations, (conversation) =>
+                _.isEqual(
+                    _.orderBy(_.map(conversation.participants, (p) => p.userId), _.identity, ['desc']),
```
Use _.sortBy instead of orderBy

ruscoder commented on src/libs/state.tsx:
```diff
@@ -1,42 +1,41 @@
-        RA.isTrue,
-        R.values(
-            R.mapObjIndexed((oldProp, key: string) => {
+    return !_.every(
```
Re-write it

Commit:
Refactor lodash calls

- containers/Chat/index.tsx: replace native slice method with lodash clone
- libs/state.tsx: reduce comparing code
- services/chat.ts: replace orderBy with sortBy, reduce arguments amount
